superset-frontend/src/dashboard/components/nativeFilters/FilterBar/FilterBar.tsx

```diff
@@ -161,7 +163,7 @@ const FilterBar: React.FC<FiltersBarProps> = ({
   const [visiblePopoverId, setVisiblePopoverId] = useState<string | null>(null);
 
   useEffect(() => {
-    if (filterConfigs.length === 0 && filtersOpen) {
+    if ((filterConfigs.length === 0 && filtersOpen) || editMode) {
```

Comment:
I think you're missing `editMode` in the deps on line 169. I also wonder why `filtersOpen` isn't there, should probably also be added..

Reply:
I've simplified the solution